dataflow/gpu-workers/Dockerfile

```diff
@@ -12,22 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM tensorflow/tensorflow:2.4.0-gpu
+# Make sure the CUDA and cuDNN versions are compatible with your TensorFlow version.
+#   https://www.tensorflow.org/install/source#gpu
+FROM nvidia/cuda:11.0.3-cudnn8-runtime-ubuntu20.04
+
+ARG python_version=3.8
 
 WORKDIR /root
 
-# Installing the requirements here makes the workers start faster since they
-# don't need to install the requirements at runtime.
-#   ℹ️ Make sure your requirements.txt includes `apache-beam[gcp]`.
+COPY --from=apache/beam_python3.8_sdk:2.28.0 /opt/apache/beam /opt/apache/beam
 COPY requirements.txt .
 
-# Don't install `tensorflow` since the base image already comes with it.
-RUN egrep -v '^tensorflow([=<>]|$)' requirements.txt > /tmp/requirements.txt \
-    && pip install --no-cache-dir -U pip \
-    && pip install --no-cache-dir -r /tmp/requirements.txt
+ENV PATH=$PATH:/opt/conda/bin
 
-# Copy the Apache Beam required files from the Beam Python SDK image.
-COPY --from=apache/beam_python3.6_sdk:2.28.0 /opt/apache/beam /opt/apache/beam
+RUN apt-get update \
+    && apt-get install -y wget \
+    && rm -rf /var/lib/apt/lists/* \
+    # The nvidia image doesn't come with Python pre-installed.
+    # We use Miniconda to install the Python version of our choice.
+    && wget -q https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+    && sh Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda \
+    && rm Miniconda3-latest-Linux-x86_64.sh \
+    # Create a new Python environment and install our requirements.
+    # We don't need to update $PATH since /usr/local is already in $PATH.
+    && conda create -y -p /usr/local python=$python_version pip \
+    && pip install --no-cache-dir -r requirements.txt \
+    && conda clean -y --all --force-pkgs-dirs
 
 # Set the entrypoint to Apache Beam SDK worker launcher.
```

Comment:
I think we need an override for pip variable or a symlink for pip (unless it's Beam 2.30.0 and higher).

Reply:
Smth like:
```
# Beam SDK worker looks for pip at a different location by default. Can be omitted with 2.30.0 and higher versions of Beam.
RUN ln -s $(which pip) /usr/local/bin/pip
```

Reply:
Thanks, added